core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -176,15 +176,29 @@ class SocketServerTest extends JUnitSuite {
 
   @Test
   def simpleRequest() {
-    val plainSocket = connect(protocol = SecurityProtocol.PLAINTEXT)
+    val plainSocket = connect()
     val serializedBytes = producerRequestBytes()
 
     // Test PLAINTEXT socket
     sendRequest(plainSocket, serializedBytes)
-    processRequest(server.requestChannel)
+    processRequest(server.dataPlaneRequestChannel)
     assertEquals(serializedBytes.toSeq, receiveResponse(plainSocket).toSeq)
   }
 
+  @Test
+  def testControlPlaneRequest(): Unit = {
```

Comment:
I'm not sure what exactly is being validated in this test. Can you clarify?

Reply:
This test is configuring the controlPlaneListener and making sure that a request that is sent to the control plane is indeed handled by the control plane, in the code.
The request should go to controlPlaneRequestChannel and not to the dataPlaneRequestChannel, which is what this is doing : 
`receiveRequest(server.controlPlaneRequestChannelOpt.get)`